### ISSUE TITLE
[Bug]: could not load library "C:/Program Files/PostgreSQL/15/lib/timescaledb-2.11.1.dll": The specified module could not be found.

### ISSUE THREAD

**Issue by draustin** [timescale/timescaledb#5879]
### What type of bug is this?

Configuration

### What subsystems and features are affected?

Configuration

### What happened?

Creating the extension results in the error `could not load library "C:/Program Files/PostgreSQL/15/lib/timescaledb-2.11.1.dll": The specified module could not be found.`.

I am aware of #5631 . I tried adding `C:\Program Files\Git\mingw64\bin` to the path (both User and System). I also tried copying `libssl-1_1-x64.dll` and `libcrypto-1_1-x64.dll` to the PostgreSQL lib folder. Finally I tried installing OpenSSL light from `https://slproweb.com/products/Win32OpenSSL.html`. None of these worked.

### TimescaleDB version affected

2.11.1

### PostgreSQL version used

15.3

### What operating system did you use?

Windows 11 x64

### What installation method did you use?

Other

### What platform did you run on?

On prem/Self-hosted

### Relevant log output and stack trace

```bash
CREATE EXTENSION IF NOT EXISTS timescaledb;

 could not load library "C:/Program Files/PostgreSQL/15/lib/timescaledb-2.11.1.dll": The specified module could not be found.
```

### How can we reproduce the bug?

```bash
Attempt to install TimescaleDB on/with the above platform/version.
```

**Comment by Greg-Smulko** [comment]
I encountered the same problem, but these steps helped:

(based on https://docs.timescale.com/self-hosted/latest/install/installation-windows/ )

Make sure you have [OpenSSL 1.1.1](https://download.firedaemon.com/FireDaemon-OpenSSL/openssl-1.1.1u.zip) installed and added to the PATH.

Make sure you have `C:\Program Files\Git\mingw64\bin` added to the PATH as per [this](https://github.com/timescale/timescaledb/issues/5631).

Restart Postgres by running `pg_ctl -D "C:\Program Files\PostgreSQL\15\data" restart` .

**Comment by akuzm** [comment]
@draustin did the suggestion by @Greg-Smulko work?

Alternatively, can you try running this thing on the timescaledb DLL to see what dependency exactly it is missing: http://www.dependencywalker.com/

**Comment by draustin** [comment]
I tried @Greg-Smulko 's suggestion but it didn't work.
I used the dumpbin program in Visual Studio to find the dependencies.

```
C:\Program Files\PostgreSQL\15\lib>dumpbin /dependents timescaledb-2.11.1.dllMicrosoft (R) COFF/PE Dumper Version 14.36.32537.0Copyright (C) Microsoft Corporation.  All rights reserved.

Dump of file timescaledb-2.11.1.dll

File Type: DLL

  Image has the following dependencies:

libssl-1_1-x64.dll
libcrypto-1_1-x64.dll
postgres.exe
WS2_32.dll
VERSION.dll
VCRUNTIME140.dll
api-ms-win-crt-string-l1-1-0.dll
api-ms-win-crt-utility-l1-1-0.dll
api-ms-win-crt-convert-l1-1-0.dll
api-ms-win-crt-runtime-l1-1-0.dll
api-ms-win-crt-math-l1-1-0.dll
api-ms-win-crt-stdio-l1-1-0.dll
KERNEL32.dll
```
At the command line, I went through the list checking that each was on my path using the `where` command. All of the DLLs starting with `api-` are missing. All others are present.

I found [hints](https://answers.microsoft.com/en-us/windows/forum/all/api-ms-win-crt-string-l1-1-0dll-is-missing-in/bb0b23b8-83ec-4fd8-b912-23c81351b5e7) that the `api` DLLs are part of Microsoft Universal C Runtime. From the [installer](https://support.microsoft.com/en-us/topic/update-for-universal-c-runtime-in-windows-322bf30f-4735-bb94-3949-49f5c49f4732)  
website I don't understand why I'm missing it - it seems to imply that it should be present for Windows 10 and later (I have Windows 11).

**Comment by akuzm** [comment]
Can you please try installing the latest Visual Studio Redistributable version? 

https://aka.ms/vs/17/release/vc_redist.x64.exe

**Comment by draustin** [comment]
@akuzm thanks for the suggestion but it didn't work. The installation of Visual Studio Redistributable seems to be fine - it shows up in Windows Control Panel. The PostgreSQL error message is the same. The DLLS beginning with `api` are still missing.

**Comment by bluppfisk** [comment]
I encounter a similar issue when trying to install on Windows server 1809 (in a docker container). Only difference is for me it says "the specified **routine** could not be found". I'm not very familiar with Windows dependencies, and forgive me if this isn't helpful, but could it be that your error message actually means it cannot find the timescaledb dll, and mine that it cannot find the required dependencies?

**Comment by jnidzwetzki** [comment]
@draustin Microsoft offers two different packages:

* the _Universal C Runtime in Windows_ (which is not available for Windows 10 and later) and 
* the _Visual C++ Redistributable for Visual Studio 2015_ (see https://www.microsoft.com/en-us/download/details.aspx?id=48145). 

The Visual C++ Redistributable needs to be installed before TimescaleDB can be installed (see [our documentation](https://docs.timescale.com/self-hosted/latest/install/installation-windows/#installing-self-hosted-timescaledb-on-windows-based-systems)). After you have installed this package, the dependencies should be available on your system.

**Comment by jnidzwetzki** [comment]
@bluppfisk It seems to be another problem causing the behavior on your system. Could you open a new issue with the steps that are needed to reproduce the error message you are facing?

**Comment by bluppfisk** [comment]
@jnidzwetzki I've solved the problem for my part (was trying to build a docker image with postgres/postgis/timescaledb on a windows server nano base image). Most likely I wasn't copying all the required DLLs (not knowing which ones I actually need). In the end I ended up copying all the msvc*.dll, vcr*.dll, openssl-1.1/x64/bin/*.dll, and openssl-1.1/x64/lib/*.lib files. That's probably too much but it now works.
